src/canvas/graphic.ts

```diff
@@ -159,7 +160,7 @@ function brushPath(ctx: CanvasRenderingContext2D, el: Path, style: PathStyleProp
         }
     }
 
-    let lineDash = style.lineDash;
+    let lineDash = normalizeLineDash(style.lineDash, style.lineWidth);
```

Comment:
Only normalize when `lineDash` exists. Or it will always enter https://github.com/ecomfe/zrender/blob/cf5e31670d4ca914b6f24dc40071e3ad6d43e23f/src/canvas/graphic.ts#L173 branch, which is an unnecessary extra cost.